ext/tests/post_hook_type_error.phpt

```diff
@@ -0,0 +1,29 @@
+--TEST--
+Check if type error in post hook is handled
+--EXTENSIONS--
+opentelemetry
+--FILE--
+<?php
+class Foo
+{
+  public function bar(): void
+  {
+    var_dump('bar');
+  }
+}
+
+\OpenTelemetry\Instrumentation\hook(
+    Foo::class,
+    'bar',
+    fn() => var_dump('pre'),
+    fn(string $scope, array $params, mixed $returnValue, ?Throwable $throwable) => var_dump('post')); //NB invalid type for $scope
+
+try {
+  (new Foo())->bar();
```

Comment:
Perhaps there should be some output after this call that is matched in expected output, to demonstrate that nothing was thrown (as the exception should be suppressed)?